pkg/network/cluster_config.go

```diff
@@ -0,0 +1,109 @@
+package network
+
+import (
+	"net"
+
+	configv1 "github.com/openshift/api/config/v1"
+	netopv1 "github.com/openshift/cluster-network-operator/pkg/apis/networkoperator/v1"
+	iputil "github.com/openshift/cluster-network-operator/pkg/util/ip"
+
+	"github.com/pkg/errors"
+)
+
+func ValidateClusterConfig(clusterConf configv1.NetworkSpec) error {
+	// Check all networks for overlaps
+	pool := iputil.IPPool{}
+
+	if len(clusterConf.ServiceNetwork) != 1 {
+		// Right now we only support a single service network
+		return errors.Errorf("spec.serviceNetwork must have length 1")
+	}
+	for _, snet := range clusterConf.ServiceNetwork {
+		_, cidr, err := net.ParseCIDR(snet)
+		if err != nil {
+			return errors.Wrapf(err, "could not parse spec.serviceNetwork %s", snet)
+		}
+		if err := pool.Add(*cidr); err != nil {
+			return err
+		}
+	}
+
+	// validate clusternetwork
+	// - has an entry
+	// - it is a valid ip
+	// - has a reasonable cidr
+	// - they do not overlap and do not overlap with the service cidr
+	for _, cnet := range clusterConf.ClusterNetwork {
+		_, cidr, err := net.ParseCIDR(cnet.CIDR)
+		if err != nil {
+			return errors.Errorf("could not parse spec.clusterNetwork %s", cnet.CIDR)
+		}
+		size, _ := cidr.Mask.Size()
+		// The comparison is inverted; smaller number is larger block
+		if cnet.HostPrefix < uint32(size) {
+			return errors.Errorf("hostPrefix %d is larger than its cidr %s",
```

Comment:
To match origin's validation, this should be `<=` not `<`. Also it checks that `HostSubnetLength >= 2` aka `HostPrefix <= 30`

Reply:
By my reading, it seems to allow a HostPrefix that is the same size as the CIDR? I'm looking [here](https://github.com/openshift/origin/blob/master/pkg/network/apis/network/validation/validation.go#L94)

Reply:
huh... somehow i looked at that same code and got the wrong answer.